types/index.d.ts

```diff
@@ -229,6 +235,125 @@ declare module '@azure/functions' {
      * Possible values for an HTTP Request user type
      */
     export type HttpRequestUserType = 'AppService' | 'StaticWebApps';
+    /**
+     * HTTP response object
```

Comment:
I think we should copy these 4-5 properties to both `HttpResponseApi` and `HttpResponseObject` and get rid of the base class. I just worry we're going to confuse people with too many `HttpResponse...` interfaces and I don't think users should ever directly use this one.

IMO we should treat this file more as "docs" than "source code", meaning readability can outweigh "NO COPYING CODE".

Reply:
Yeah that makes sense. Do you think it'd be helpful to not export the base interface so that users can't directly use it? Or is it more about the readability of this file itself?

Reply:
What do you mean "not export the base interface"? I don't think we need it at all

Reply:
Yes I was talking about the readability of this file. I use "go to definition" and read typescript definition files all the time. I basically prefer those over online docs. Plus if the file itself is readable, that usually means the intellisense works better

Reply:
Cool yeah makes sense.